ArtOfIllusion/src/artofillusion/script/ExecuteScriptWindow.java

```diff
@@ -30,22 +38,97 @@ public class ExecuteScriptWindow extends BFrame
+        java.util.List<String> newRecentFiles = new ArrayList<String>();
+        newRecentFiles.add (filePath);
+        for (String recentFile : recentFiles) 
+                    if (!recentFile.equals(filePath)) // If the current file already has a timestamp it will be updated below
+                        newRecentFiles.add (recentFile);
+        pref.put("recentFiles", String.join (File.pathSeparator, 
+                (String []) newRecentFiles.subList(0, 10).toArray(new String[0])));
+    }
+    
+    public static String [] getRecentScripts()
+    {
+        final Preferences pref = Preferences.userNodeForPackage(ExecuteScriptWindow.class);
+        final java.util.List<String> recentScripts = Arrays.asList (pref.get("recentFiles", "").split(File.pathSeparator));
+        return recentScripts.toArray(new String [0]);
+    } 
+
+  /** This is used to track the "changed" status of the script being edited. */
+  private class ScriptKeyListener implements KeyListener {
+
+      @Override
+      public void keyTyped(KeyEvent e) {
+          save.setEnabled(true);        
+      }
+
+      @Override
+      public void keyPressed(KeyEvent e) {
+      }
+
+      @Override
+      public void keyReleased(KeyEvent e) {
+      }
+        
+    }
+    
+  /**
+   * 
+   * @param win
+   * @param scriptAbsolutePath {@link ExecuteScriptWindow#NEW_SCRIPT_NAME} if this is a new script
+   * @param scriptLanguage May be {@link ScriptRunner#UNKNOWN_LANGUAGE} if this is a new script
+   */
+  public ExecuteScriptWindow(LayoutWindow win, String scriptAbsolutePath, String scriptLanguage) throws IOException
   {
-    super(lastScriptName);
-    scriptName = lastScriptName;
+    super(scriptAbsolutePath);
+    setScriptNameFromFile(scriptAbsolutePath);
+    language = scriptLanguage;
+    scriptPath = scriptAbsolutePath;
+     // Get the extensions dynamically
+      final java.util.List <String> extensions = new ArrayList <String>();
+      for (String language : ScriptRunner.getLanguageNames ()) {
+          extensions.add (ScriptRunner.getFilenameExtension(language));
+      }
+      scriptFileFilter = new javax.swing.filechooser.FileNameExtensionFilter(
+        "Script files", (String[]) extensions.toArray(new String [0]));
+
+    // TODO FIXME the edit zone does not maximize when the panel is maximized
+    // TODO FIXME Same problem when the window is put to the side
     BorderContainer content = new BorderContainer();
     setContent(content);
     window = win;
-    if (scriptDir == null)
-      scriptDir = new File(ArtOfIllusion.TOOL_SCRIPT_DIRECTORY);
-    scriptText = new RSyntaxTextArea(lastScript, 25, 100);
+    String editorTextContent = "";
+    if (scriptLanguage != ScriptRunner.UNKNOWN_LANGUAGE && scriptAbsolutePath.contains(".")) {
+        editorTextContent = ArtOfIllusion.loadFile(new File (scriptAbsolutePath));
+    }
+    scriptText = new RSyntaxTextArea(editorTextContent, 25, 100);
```

Comment:
I think that we may have to change the layout used by the `BorderContainer`

Reply:
This looks like a bug to me, because this strange behaviour appears depending on whether or not scrollbars are visible in the text area : try alternating Windows-Right, Windows-Up, Windows-Down with different files and line lengths, you'll see it (I hope this behaviour is the same on all platforms..). 
And also, the scrollbars sometimes do not appear when they should. I have absolutely no idea where to look for for the root cause of the bug : RSyntaxTextArea, Buoy, AOI? Layout, as you mentioned, events, scrollbars? 
I'd be willing to let this bug live for now and let someone who is more familiar with the code than I am hunt it, as it's not really critical nor blocking - apparenty no one had realized its presence until now..?

Reply:
And I was wrong. Somehow, the maximize (and restore) events are not triggering as a `WindowResizedEvent` repaint on the `RSyntaxTextArea` - or more probably the `RTextScrollPane` or `AWTWidget` that it is wrapped in.

This has probably not been noticed because most of us don't actually use the script windows maximized, and manually re-sizing the window *does* work correctly. Clicking in, or attempting to scroll the script pane also triggers a repaint.

Reply:
Now has it's own issue: #283